src/main/java/cz/cvut/kbss/termit/service/document/backup/DocumentBackupManager.java

```diff
@@ -39,11 +43,26 @@
 @Service
 public class DocumentBackupManager {
     private static final Logger LOG = LoggerFactory.getLogger(DocumentBackupManager.class);
-
+    private static final Map<ResourceSaveReason, BackupReason> SAVE_REASONS_FOR_BACKUP = Map.of(
+            ResourceSaveReason.NEW_OCCURRENCE, BackupReason.NEW_OCCURRENCE,
+            ResourceSaveReason.REMOVE_OCCURRENCE, BackupReason.REMOVE_OCCURRENCE,
+            ResourceSaveReason.REUPLOAD, BackupReason.REUPLOAD
+    );
     private final Path storageDirectory;
+    private final ResourceRepositoryService resourceRepositoryService;
 
-    public DocumentBackupManager(Configuration config) {
+    public DocumentBackupManager(Configuration config, ResourceRepositoryService resourceRepositoryService) {
         this.storageDirectory = Path.of(config.getFile().getStorage());
+        this.resourceRepositoryService = resourceRepositoryService;
+    }
+
+    /**
+     * Maps {@link ResourceSaveReason} to {@link BackupReason} if the backup should be created.
+     * @param saveReason The save reason to map
+     * @return The mapped value or empty optional if the backup should not be created.
+     */
+    public static Optional<BackupReason> mapSaveReasonToBackup(ResourceSaveReason saveReason) {
```

Comment:
Why is the mapping not part of one of the enums? For example, constants in `BackupReason` could have a corresponding `ResourceSaveReason` (if applicable).